src/rpc/net.cpp

```diff
@@ -65,12 +65,80 @@ UniValue ping(const JSONRPCRequest& request)
+    obj.push_back(Pair("relaytxes", stats.fRelayTxes));
+    obj.push_back(Pair("lastsend", stats.nLastSend));
+    obj.push_back(Pair("lastrecv", stats.nLastRecv));
+    obj.push_back(Pair("bytessent", stats.nSendBytes));
+    obj.push_back(Pair("bytesrecv", stats.nRecvBytes));
+    obj.push_back(Pair("conntime", stats.nTimeConnected));
+    obj.push_back(Pair("timeoffset", stats.nTimeOffset));
+    if (stats.dPingTime > 0.0) {
+        obj.push_back(Pair("pingtime", stats.dPingTime));
+    }
+    if (stats.dMinPing < static_cast<double>(std::numeric_limits<int64_t>::max())/1e6) {
+        obj.push_back(Pair("minping", stats.dMinPing));
+    }
+    if (stats.dPingWait > 0.0) {
+        obj.push_back(Pair("pingwait", stats.dPingWait));
+    }
+    obj.push_back(Pair("version", stats.nVersion));
+    // Use the sanitized form of subver here, to avoid tricksy remote peers from
+    // corrupting or modifying the JSON output by putting special characters in
+    // their ver message.
+    obj.push_back(Pair("subver", stats.cleanSubVer));
+    obj.push_back(Pair("inbound", stats.fInbound));
+    obj.push_back(Pair("addnode", stats.m_manual_connection));
+    obj.push_back(Pair("startingheight", stats.nStartingHeight));
+    if (got_state_stats) {
+        obj.push_back(Pair("banscore", state_stats.nMisbehavior));
+        obj.push_back(Pair("synced_headers", state_stats.nSyncHeight));
+        obj.push_back(Pair("synced_blocks", state_stats.nCommonHeight));
+        UniValue heights(UniValue::VARR);
+        for (int height : state_stats.vHeightInFlight) {
+            heights.push_back(height);
+        }
+        obj.push_back(Pair("inflight", heights));
+    }
+    obj.push_back(Pair("whitelisted", stats.fWhitelisted));
+
+    UniValue bytes_sent_per_message(UniValue::VOBJ);
+    for (const mapMsgCmdSize::value_type &i : stats.mapSendBytesPerMsgCmd) {
+        if (i.second > 0) {
+            bytes_sent_per_message.push_back(Pair(i.first, i.second));
+        }
+    }
+    obj.push_back(Pair("bytessent_per_msg", bytes_sent_per_message));
+
+    UniValue bytes_recvd_per_message(UniValue::VOBJ);
+    for (const mapMsgCmdSize::value_type &i : stats.mapRecvBytesPerMsgCmd) {
+        if (i.second > 0) {
+            bytes_recvd_per_message.push_back(Pair(i.first, i.second));
+        }
+    }
+    obj.push_back(Pair("bytesrecv_per_msg", bytes_recvd_per_message));
+}
+
 UniValue getpeerinfo(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() != 0)
+    if (request.fHelp || request.params.size() > 1)
         throw std::runtime_error(
             "getpeerinfo\n"
-            "\nReturns data about each connected network node as a json array of objects.\n"
+            "\nReturns data about connected network nodes as a json array of objects.\n"
+            "\nArguments:\n"
+            "1. \"id\"         (numeric, optional) Only return information about the peer with the specified id.\n"
```

Comment:
Nit: No period at end (seems to be the norm). Also add this to the first line (`"getpeerinfo \"id\"\n"`)

Reply:
fixed